src/MailThief.php

```diff
@@ -101,7 +101,7 @@ public function later($delay, $view, array $data, $callback, $queue = null)
 
     public function hasMessageFor($email)
     {
-        return $this->messages->contains(function ($i, $message) use ($email) {
+        return $this->messages->contains(function ($i, Message $message) use ($email) {
```

Comment:
Hm. I wouldn't call it suppression of a warning, but rather "documentation via typehint." 

However, I don't mind documentation via typehint, as long as it doesn't negatively impact the clarity and performance and flexibility of the code.

I can't currently see how this would have a negative impact, so let's do it. We can always take it out if it causes us any trouble.